test/AccountManager.test.ts

```diff
@@ -0,0 +1,292 @@
+import type { EnvelopingTypes } from '@rsksmart/rif-relay-contracts/dist/typechain-types/contracts/RelayHub';
+import { expect, use } from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import { providers, Wallet } from 'ethers';
+import { createSandbox } from 'sinon';
+import sinonChai from 'sinon-chai';
+import AccountManager from '../src/AccountManager';
+import * as typedDataUtils from '../src/typedRequestData.utils';
+
+use(sinonChai);
+use(chaiAsPromised);
+
+const sandbox = createSandbox();
+
+const createRandomAddress = () => Wallet.createRandom().address;
+
+describe('AccountManager', function () {
+  describe('constructor', function () {
+    after(function () {
+      sandbox.restore();
+    });
+    afterEach(function () {
+      sandbox.restore();
+    });
+
+    it('should store signer and chain id', function () {
+      const expectedProvider: providers.Provider = sandbox.createStubInstance(
+        providers.JsonRpcProvider
+      );
+      const expectedChainId = Math.random() * 100;
+      const accountManager = new AccountManager(
+        expectedProvider,
+        expectedChainId
+      );
+
+      expect(
+        (accountManager as unknown as { _provider: providers.Provider })
+          ._provider,
+        'Provider'
+      ).to.equal(expectedProvider);
+      expect(accountManager.chainId, 'Chain ID').to.equal(expectedChainId);
+    });
+
+    describe('', function () {
+      let deployRequest: EnvelopingTypes.DeployRequestStruct;
+      let relayRequest: EnvelopingTypes.RelayRequestStruct;
+
+      let accountManager: AccountManager;
+      let stubProvider: providers.JsonRpcProvider;
+
+      beforeEach(function () {
+        const relayOrDeployRequest = {
```

Comment:
if we are stubbing the methods, why do we need the whole object? I try just to use the necessary properties and for ce the conversion to the required object.

Reply:
Could you please clarify as to why you prefer to use only the necessary objects for given test?
I'd argue that:
1. it's better to build your objects that do not require change once (which, fair enough, is not what I am doing here, but probably should)
2. here specifically, it allowed me to split and see what is actually different between relay and deploy requests
3. by default you want to pass a complete object and if you want test for a lack of a property you remove it
4. reduces the amount of constants you need to then always build your object

Reply:
There is no specific reason. Just trying to follow some standard. Ill change my test to use the whole object object.

Reply:
ok so resolve this comment please if there is nothing to follow up on

Reply:
btw I moved it above the main describe block